src/components/rules/Rules.jsx

```diff
@@ -0,0 +1,47 @@
+import { useLocation, useNavigate } from "react-router-dom";
+import { useAuth, useQuiz } from "../../contexts";
+
+export const Rules = ({ quizId, rulesAccepted }) => {
+  const { getQuizQuestions } = useQuiz();
+  const { isUserLoggedIn } = useAuth();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  return (
+    <div className="list-container">
+      <div className="list-heading">Accept the rules to continue</div>
+      <ul className="list">
+        <li className="list-item">
+          1. There will be 4 questions with 4 options each
+        </li>
+        <li className="list-item">
+          2. To submit the quiz answer all questions
+        </li>
+        <li className="list-item">
+          3. Each correct answer will be awarded
+          <span className="positive-marking"> +5 </span> points
+        </li>
+        <li className="list-item">
+          4. Each wrong answer will be awarded
+          <span className="negative-marking"> -1 </span> points
+        </li>
+        <li className="list-item">
+          5. Quitting the quiz in between will reset the quiz
+        </li>
+      </ul>
+      <button
+        className="btn btn-secondary"
+        onClick={() => {
+          if (isUserLoggedIn) {
+            getQuizQuestions(quizId);
+            rulesAccepted(true);
+          } else {
+            navigate("/login", { state: { from: location }, replace: true });
+          }
```

Comment:
I think you can put this whole onClick function in another function and call that function here, which is much more readable

Reply:
Done